libsolidity/analysis/TypeChecker.cpp

```diff
@@ -351,6 +351,9 @@ bool TypeChecker::visit(FunctionDefinition const& _function)
 	if (_function.overrides() && _function.isFree())
 		m_errorReporter.syntaxError(1750_error, _function.location(), "Free functions cannot override.");
 
+	if (!_function.modifiers().empty() && _function.isFree())
+		m_errorReporter.syntaxError(6261_error, _function.location(), "Free functions cannot have modifiers.");
+
```

Comment:
This change looks like it doesn't belong here. Is this something that should be in #9364?

Reply:
Indeed, I was just about to remove that